client/src/components/cart/CartTable.js

```diff
@@ -0,0 +1,187 @@
+import { useContext } from "react";
+import styled from "styled-components";
+import Payment from "../forms/Payment";
+import { Link } from "react-router-dom";
+import { CartContext } from "../context/Context";
+function CartTable() {
+  const [cart, setCart] = useContext(CartContext);
+  const toggleItemQuantity = (itemName, subtract) => {
+    let newState;
+    if (subtract) {
+      newState = cart.map((item) => {
+        item.name === itemName && item.quantity--;
+        return item;
+      });
+    } else {
+      newState = cart.map((item) => {
+        item.name === itemName && item.quantity++;
+        return item;
+      });
+    }
+    setCart(newState);
+  };
+  const handleDelete = (name) => {
+    const filteredCart = cart.filter((item) => item.name != name);
+    setCart(filteredCart);
+  };
+  return (
+    <TableContainer>
+      <h2>Your Cart</h2>
+      {cart.length ? (
```

Comment:
I'd say first check if `cart` is defined before calling `.length` on it. E.g `cart && cart.length`, just to be safe.

Reply:
That makes sense, I'll do that.